File: core/src/main/java/forpleuvoir/dhwu1a/core/message/FriendMessage.java
package forpleuvoir.dhwu1a.core.message;

import com.google.gson.JsonObject;
import com.google.gson.annotations.SerializedName;
import forpleuvoir.dhwu1a.core.message.base.Message;
import forpleuvoir.dhwu1a.core.message.base.MessageType;
import forpleuvoir.dhwu1a.core.user.Friend;

import static forpleuvoir.dhwu1a.core.common.ApiKey.FRIEND;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.message
 * <p>#class_name FriendMessage
 * <p>#create_time 2021/7/1 1:29
 */
public class FriendMessage extends Message {
    /**
     * 发送消息的好友
     */
    @SerializedName(FRIEND)
    public final Friend friend;

    public FriendMessage(JsonObject object) {
        super(MessageType.FriendMessage, object);
        this.friend = (Friend) sender.getUser();
    }

    @Override
    public void printMessageLog() {
        log.info("RM/Bot.{}:{}({}) >> {}", bot.id, friend.data.nickname, friend.id, this.toPlainText());
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/message/FriendMessage.kt
package forpleuvoir.dhwu1a.core.message

import com.google.gson.JsonObject
import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.FRIEND
import forpleuvoir.dhwu1a.core.message.base.Message
import forpleuvoir.dhwu1a.core.message.base.MessageType
import forpleuvoir.dhwu1a.core.user.Friend

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.message
 *
 * #class_name FriendMessage
 *
 * #create_time 2021/7/1 1:29
 */
class FriendMessage(`object`: JsonObject) : Message(MessageType.FriendMessage, `object`) {
    /**
     * 发送消息的好友
     */
    @JvmField
    @SerializedName(FRIEND)
    val friend: Friend? = sender.user as Friend?
    override fun printMessageLog() {
        log.info("RM/Bot.{}:{}({}) >> {}", bot.id, friend?.data?.nickname, friend?.id, this.toPlainText())
    }

}